chia/wallet/wallet_state_manager.py

```diff
@@ -1745,9 +1739,11 @@ async def convert_puzzle_hash(self, wallet_id: uint32, puzzle_hash: bytes32) ->
 
         return puzzle_hash
 
-    def get_dl_wallet(self) -> Optional[DataLayerWallet]:
-        for _, wallet in self.wallets.items():
+    def get_dl_wallet(self) -> DataLayerWallet:
+        for wallet in self.wallets.values():
             if wallet.type() == WalletType.DATA_LAYER.value:
-                assert isinstance(wallet, DataLayerWallet)
+                assert isinstance(
+                    wallet, DataLayerWallet
+                ), f"WalletType.DATA_LAYER should be a DataLayerWallet instance got: {type(wallet).__name__}"
                 return wallet
-        return None
+        raise ValueError("DataLayerWallet not available")
```

Comment:
Could afford to be it's own error, but sure, this fits with existing patterns.

Reply:
Yeah i just moved it into there from all the outer places, we use the `ValueError` in a bunch of other related places so thats a separate PR.